tests/magic_spell_test.cpp

```diff
@@ -0,0 +1,105 @@
+#include "magic.h"
+
+#include "catch/catch.hpp"
+
+// Return experience points needed to level up a spell, starting at from_level
+static int spell_xp_to_level( const spell_id sp_id, const int from_level )
+{
+    spell test_spell( sp_id );
+    test_spell.set_level( from_level );
+    return test_spell.exp_to_next_level();
+}
+
+TEST_CASE( "experience to level up spell", "[magic][spell][xp]" )
+{
+    spell_id lava_id( "test_spell_lava" );
+
+    SECTION( "experience needed to level up spells" ) {
+        CHECK( spell_xp_to_level( lava_id, 0 ) == 4881 );
+        CHECK( spell_xp_to_level( lava_id, 1 ) == 1751 );
+        CHECK( spell_xp_to_level( lava_id, 2 ) == 2027 );
+        CHECK( spell_xp_to_level( lava_id, 3 ) == 2347 );
+        CHECK( spell_xp_to_level( lava_id, 4 ) == 2717 );
+        CHECK( spell_xp_to_level( lava_id, 5 ) == 3147 );
+        CHECK( spell_xp_to_level( lava_id, 6 ) == 3644 );
+        CHECK( spell_xp_to_level( lava_id, 7 ) == 4220 );
+        CHECK( spell_xp_to_level( lava_id, 8 ) == 4886 );
+        CHECK( spell_xp_to_level( lava_id, 9 ) == 5659 );
+        CHECK( spell_xp_to_level( lava_id, 10 ) == 6552 );
+        CHECK( spell_xp_to_level( lava_id, 11 ) == 7587 );
+        CHECK( spell_xp_to_level( lava_id, 12 ) == 8785 );
+        CHECK( spell_xp_to_level( lava_id, 13 ) == 10173 );
+        CHECK( spell_xp_to_level( lava_id, 14 ) == 11780 );
+        CHECK( spell_xp_to_level( lava_id, 15 ) == 13641 );
+        CHECK( spell_xp_to_level( lava_id, 16 ) == 15795 );
+        CHECK( spell_xp_to_level( lava_id, 17 ) == 18291 );
+        CHECK( spell_xp_to_level( lava_id, 18 ) == 21180 );
+        CHECK( spell_xp_to_level( lava_id, 19 ) == 24525 );
+    }
+}
+
+// Return spell damage at a given level
+static int spell_damage( const spell_id test_spell_id, const int spell_level )
+{
+    spell test_spell( test_spell_id );
+    test_spell.set_level( spell_level );
+    return test_spell.damage();
+}
+
+TEST_CASE( "spell damage", "[magic][spell][damage]" )
+{
+    spell_id sp_id( "test_spell_pew" );
+    spell_type sp_type = sp_id.obj();
```

Comment:
```suggestion
    const spell_type &sp_type = sp_id.obj();
```
might as well make this const, since you're not modifying it.